src/main/scala/regmapper/RegField.scala

```diff
@@ -6,18 +6,30 @@ import Chisel._
 import chisel3.util.{ReadyValidIO}
 
 import freechips.rocketchip.util.{SimpleRegIO}
+import freechips.rocketchip.util.property._
+import chisel3.internal.sourceinfo.{SourceInfo, SourceLine}
+import freechips.rocketchip.config.Parameters
 
 case class RegReadFn private(combinational: Boolean, fn: (Bool, Bool) => (Bool, Bool, UInt))
 object RegReadFn
 {
+  def coverread(fn: (Bool, Bool) => (Bool, Bool, UInt))(implicit sourceInfo: SourceInfo, p: Parameters): (Bool, Bool) => (Bool, Bool, UInt) = {
+    case (ivalid: Bool, oready: Bool) => {
+      val (iready, ovalid, data) = fn(ivalid, oready)
+      cover(iready && ivalid, "regread_in", "RegField Read Request Initiate")
```

Comment:
Why not regread_start/finish instead of in/out, to match the longer description?